### PR TITLE
docs: Add `mkdocs` support to render static documentation site on GitHub

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add `mkdocs` support to render static documentation site on GitHub

## Motivation and Context
- Working on improving the documentation and making it easier to both present and find information for users

Right now this is bare bones just to make sure the CI flow gets setup correctly before we start porting over the existing docs and adding new

Example:
image

## Breaking Changes
- No

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: docs/assets/terafformaws.png
Comment (antonbabenko) [inline docs/assets/terafformaws.png:1]: Double "ffffffff" in the file name :)

**Review by antonbabenko — COMMENTED** [review]
Great idea. I use mkdocs also for my other things, and it is great.

BTW, maybe AWS has special access to some of the commercial features available for Sponsors-only of mkdocs. I am not sure who to ask. This is just an idea to research if necessary.

File: docs/assets/terafformaws.png
Comment (bryantbiggs) [reply docs/assets/terafformaws.png:1]: really messed up the name on that one! - fixed in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/3070/commits/4945ca22af45ecb5921c7b11f25fe2a1b5fc6a22

> BTW, maybe AWS has special access to some of the commercial features available for Sponsors-only of mkdocs

Oh, I didn't know there were commercial features. I can ask around amongst some of the OSPO folks to see

**Comment by bryantbiggs** [comment]
@antonbabenko let me know if there is anything else on this one - with this one I was just setting up the initial site and ensuring the CI auto deploy works, etc. before I start moving docs. I should have a follow up PR today for all of the content (hopefully, we'll see how far I get 😅 )

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 20.15.0 :tada:

**Pull request merged**